cmd/nerdctl/login.go

```diff
@@ -176,12 +176,20 @@ func GetDefaultAuthConfig(checkCredStore bool, serverAddress string, isDefaultRe
 
 func loginClientSide(ctx context.Context, cmd *cobra.Command, auth types.AuthConfig) (string, error) {
 	host := auth.ServerAddress
-	if strings.Contains(host, "://") {
-		u, err := url.Parse(host)
+
+	u, err := url.Parse(host)
+	if err != nil {
+		return "", err
+	}
+
+	if u.Host == "" {
+		u, err := url.Parse("https://" + host)
 		if err != nil {
 			return "", err
 		}
 		host = u.Host
+	} else {
+		host = u.Host
```

Comment:
Can we have a unit test and/or comment lines to clarify the intention this code?